mailpoet/lib/Cron/DaemonActionSchedulerRunner.php

```diff
@@ -123,6 +130,17 @@ public function afterProcess() {
     }
   }
 
+  /**
+   * This method is hooked into action_scheduler_maximum_execution_time_likely_to_be_exceeded
+   * and used to listen on how many execution time is needed.
+   * The execution limit is then used for the daemon run
+   */
+  public function storeRemainingExecutionLimit($likelyExceeded, $runner, $processedActions, $executionTime, $maxExecutionTime) {
+    $newLimit = floor(($maxExecutionTime - $executionTime) - self::EXECUTION_LIMIT_MARGIN);
+    $this->remainingExecutionLimit = intval(max($newLimit, 0));
```

Comment:
If the input numbers are floats then maybe `remainingExecutionLimit` could also be float? That would avoid `floor` and `intval` calls.